core/commonMain/src/implementations/persistentOrderedMap/PersistentOrderedMapBuilderContentIterators.kt

```diff
@@ -28,7 +28,10 @@ internal open class PersistentOrderedMapBuilderLinksIterator<K, V>(
         lastIteratedKey = nextKey
         nextWasInvoked = true
         index++
-        val result = builder.hashMapBuilder[nextKey]!!
+        @Suppress("UNCHECKED_CAST")
+        val result = builder.hashMapBuilder.getOrElse(nextKey as K) {
+            throw ConcurrentModificationException("Hash code of a key ($nextKey) has changed after it was added to the persistent map.")
```

Comment:
Will we get the same message for a persistent set backed by this map? Better to make it more generic then, e.g. "... to the persistent collection."

Reply:
Oh, I see the changes for set below